src/write_gcm.c

```diff
@@ -2147,6 +2169,52 @@ static monitored_resource_t *wg_monitored_resource_create(
   return NULL;
 }
 
+static monitored_resource_t *wg_monitored_resource_create_from_metadata_agent(
+    const char *metadata_agent_url, const char *project_id,
+    const char *resource_id) {
+  if (resource_id == NULL) {
+    WARNING("write_gcm: Hostname not set. Cannot query Metadata Agent.");
```

Comment:
Updated. For completeness I changed it to indicate what was expected in the Hostname along with the fact that it hasn't been set.